### PR TITLE
13983 Add nested arrays for extra_choices in CustomFieldChoiceSet 

### LINKED ISSUE #13983: Custom Field Choice Set Bulk Import error "Nested arrays must have same length"
### NetBox version

v3.6.3

### Python version

3.10

### Steps to Reproduce

1. Select Custom Field Choices import
2. Data to import:

CSV:
```
name;description;extra_choices
numbers;Numbers;7,13,18,23,26,32,38,80
```

JSON:
```
{
    "name": "numbers",
    "description": "Numbers",
    "extra_choices": [7,13,18,23,26,32,38,80]
}
```

YAML:
```
---
name: numbers
description: Numbers
extra_choices: 7,13,18,23,26,32,38,80
```

3. Click Submit

### Expected Behavior

It is expected to create the cusotm field choices

### Observed Behavior

Returns this error in all cases: Record 1 extra_choices: Nested arrays must have the same length.

### PR DESCRIPTION
**Pull request by arthanson**
### Fixes: #13983 

Relies on change in #14469 Allows to specify multiple choices and labels for CSV import of extra_choices - just quote the whole field: "choice1:First Choice,choice2:Second Choice"

### CODE REVIEW

File: netbox/extras/forms/bulk_import.py [hunk 1]
```diff
@@ -85,6 +90,17 @@ class Meta:
...
 
+    def clean_extra_choices(self):
+        if isinstance(self.cleaned_data['extra_choices'], list):
+            data = []
+            for line in self.cleaned_data['extra_choices']:
+                try:
+                    value, label = re.split(r'(?<!\\):', line, maxsplit=1)
+                except ValueError:
+                    value, label = line, line
+                data.append((value, label))
```
Comment (jeremystretch) [inline netbox/extras/forms/bulk_import.py:101]: We also need to normalize any escaped colons (`\:`) when saving the data.
```suggestion
                value = value.replace('\:', ':')
                label = label.replace('\:', ':')
                data.append((value, label))
```

(This should also be applied to the model form for #14467.)

**Review by jeremystretch — CHANGES_REQUESTED** [review]

File: netbox/extras/forms/bulk_import.py (hunk 1 above)
Comment (arthanson) [reply netbox/extras/forms/bulk_import.py:101]: It needed to be saved with the escaped character as doing the save/edit cycle would replace it.  This has a change to replace the label when it is displayed in the model form using the custom field.

Comment (jeremystretch) [reply netbox/extras/forms/bulk_import.py:101]: The escape character should never be saved to the database; it's relevant only for processing raw input. I don't follow what you mean about the save/edit cycle, but we'll have to figure out a different approach to ensure the normalized values are stored.

Comment (arthanson) [reply netbox/extras/forms/bulk_import.py:103]: The save/edit cycle meant if you edit and save a choice-set then re-edit it.  You either have to save the escape characters or else re-add them when edit.  I've changed to re-add them when you edit so they aren't saved in the database.

**Review by jeremystretch — APPROVED** [review]

**Pull request merged**